src/Conversions.ts

```diff
@@ -98,3 +103,60 @@ export const toNativeWeiFrom = async ({
         precision: RBTC_CHAIN_DECIMALS
     });
 };
+
+/**
+ * Converts gas estimation to token amount
+ * @param estimation estimation to be converted
+ * @param xRate exchange rate of the token
+ * @param gasPrice gasPrice
+ * @returns 'WEI' representation of the gas converted to token
+ */
+export const convertGasToToken = (
+    estimation: InputParam,
+    xRate: InputParam,
+    gasPrice: InputParam
+): BigNumber => {
+    const bigEstimation = new BigNumber(estimation);
+    const bigRate = new BigNumber(xRate);
+    const bigPrice = new BigNumber(gasPrice);
+    if (
+        invalidNumber(bigEstimation) ||
+        invalidNumber(bigRate) ||
+        invalidNumber(bigPrice) ||
+        bigRate.isZero()
+    ) {
+        return new BigNumber(0);
+    }
+    const total = bigEstimation.multipliedBy(bigPrice);
+    return total.dividedBy(bigRate);
+};
+
+/**
+ * Converts gas estimation to native amount
+ * @param estimation estimation to be converted
+ * @param gasPrice gasPrice
+ * @returns 'WEI' representation of the gas converted to native
+ */
+export const convertGasToNative = (
+    estimation: InputParam,
+    gasPrice: InputParam
+): BigNumber => {
+    const bigEstimation = new BigNumber(estimation);
+    const bigPrice = new BigNumber(gasPrice);
+    if (invalidNumber(bigEstimation) || invalidNumber(bigPrice)) {
+        return new BigNumber(0);
+    }
+    return bigEstimation.multipliedBy(bigPrice);
+};
+
+/**
+ * Verify of the number is not valid
+ * @param value BigNumber value
+ * @returns boolean flag
+ */
+const invalidNumber = (value: BigNumber): boolean => {
```

Comment:
hmm you sure you're not overthinking this? I understand what it is checking, but when would such situation occur? Who would send these numbers in? Surely, if there is an opportunity to do that at higher levels, this should be verified there? Otherwise we would have to validate ALL the numbers we handle, no? What am I missing?

Reply:
The way that I think about this was, that an estimation could be negative and that an exchangeRate could be zero or any number since we depends on an external system, but its kind of overthinking or validating some pretty weird scenarios.

Reply:
Although it may be unlikely to happen, it doesn't hurt